core/src/main/scala/cats/instances/map.scala

```diff
@@ -53,8 +57,33 @@ trait MapInstances extends cats.kernel.instances.MapInstances {
       def foldRight[A, B](fa: Map[K, A], lb: Eval[B])(f: (A, Eval[B]) => Eval[B]): Eval[B] =
         Foldable.iterateRight(fa.values.iterator, lb)(f)
 
+      def tailRecM[A, B](a: A)(fn: A => Map[K, A Xor B]): Map[K, B] = {
```

Comment:
did a revised version of this. Thanks (was late when I wrote the first one. Sorry).